### PR TITLE
Add graphite timeout

### PR DESCRIPTION
**Pull request by erikzaadi**

### CODE REVIEW

**Comment by erikzaadi** [comment]
@hkariti @shaharke

**Comment by hkariti** [comment]
:+1:

File: check_graphite [hunk 1]
```diff
@@ -72,9 +75,13 @@ if graphite.options.hostname:
...
 
-usock = urllib2.urlopen(graphite.options.url)
-data = usock.read()
-usock.close()
+try:
+    usock = urllib2.urlopen(graphite.options.url, timeout=timeout)
+    data = usock.read()
+    usock.close()
+except socket.timeout:
+    graphite.unknown_error("Error connecting to Graphite: Timeout of 20 seconds reached")
```
Comment (hkariti) [inline check_graphite]: 20 is still hardcoded here

File: check_graphite [hunk 2]
```diff
@@ -53,12 +54,14 @@ graphite.add_option("P", "password", "Password for authentication")
...
 graphite.add_option("n", "none", "Ignore None values: 'yes' or 'no' (default no)")
 graphite.add_option("f", "function", "Function to run on retrieved values: avg/min/max/last/sum (default 'avg')",
-  default="avg")
+graphite.add_option("t", "timeout", "timeout in seconds for accessing graphite", default="20")
 
 graphite.enable_status("warning")
 graphite.enable_status("critical")
 graphite.start()
 
+timeout = int(graphite.options.timeout)
```
Comment (hkariti) [inline check_graphite:64]: add `type=int` to add_option?

**Pull request merged**